bazaar/core/tasks.py

```diff
@@ -405,30 +405,44 @@ def ssdeep_analysis(sha256):
 
         with TemporaryDirectory() as tmp_dir:
             apk = zipfile.ZipFile(f)
-            apk.extractall(tmp_dir)
+
+            file_list = apk.namelist()
+            dex_files = []
+            for member in file_list:
+                if member.endswith('.dex'):
+                    dex_files.append(member)
+                    try:
+                        apk.extract(member, tmp_dir)
+                    except Exception as e:
+                        logging.error('Can not extract member: %s due to error %s', member, e)
+
+            logging.info('Extracted %s .dex files', len(dex_files))
+
+            apk.extract('AndroidManifest.xml')
```

Comment:
- we should handle errors on those 2 file extractions
- why there is no destination path parameter?

Reply:
Code review is the best! Thanks for spotting issues.
Fixed.